src/matrix.hpp

```diff
@@ -134,6 +136,62 @@ struct matrix {
 
     [[nodiscard]] friend constexpr auto
     operator==(const matrix& lhs, const matrix& rhs) -> bool = default;
+
+    template <Vector W, std::size_t OtherCols>
+    requires(W::size == Cols) [[nodiscard]] friend constexpr auto
+    operator*(const matrix<V, Cols>& m0, const matrix<W, OtherCols>& m1)
+        -> matrix<V, OtherCols>
+    {
+        matrix<V, OtherCols> m2{};
+        std::transform(
+            m1.begin(), m1.end(), m2.begin(), [&m0](const W& m1_column) {
+                return m0 * m1_column;
+            });
+        return m2;
+    }
+
+    template <Vector W>
+    requires(W::size == Cols) [[nodiscard]] friend constexpr auto
+    operator*(const matrix<V, Cols>& m0, const W& v1) -> V
+    {
+        return std::transform_reduce(
+            m0.begin(),
+            m0.end(),
+            v1.begin(),
```

Comment:
Yeah, transpose comes a bit later in the chain. And yes, this would be easier done having a view cycling over the rows, the the output would simply be generated as inner products of the rows and v1